src/core/shape_inference/include/lstm_cell_shape_inference.hpp

```diff
@@ -28,8 +28,8 @@ void lstm_shape_infer(const OpsType* op,
     // If rank is dynamic, then output_shape is undefined
     for (size_t i = 0; i < input_shapes.size() && i < 6; i++) {
         input_rank_static[i] = input_shapes[i].rank().is_static();
-        all_rank_dynamic &= !input_rank_static[i];
-        all_rank_static &= input_rank_static[i];
+        all_rank_dynamic = all_rank_dynamic && !input_rank_static[i];
```

Comment:
Why do we need this change? Both approaches look correct.

Reply:
Basically, it's because the ` &` is for bit-wise, but here the more precious one is the logic `AND` but not bit-wise `AND`. More precisely, The klockwork scan says `Operands of different size in bitwise operation`